bio_hansel/kmer_count/utils.py

```diff
@@ -54,17 +54,19 @@ def error_rate(hist: pd.DataFrame) -> float:
     :return:
             float value containing the error rate.
     """
-    n_singleton_kmers = float(hist[hist.freq == 2].obs)
+    n_singleton_kmers = float(hist[hist.freq == 1].obs)
     logging.debug('n_singleton_kmers %s', n_singleton_kmers)
     # get the number of unique k-mers within the observation
     #   We can simply follow the "curve of the graph" to find the number of unique k-mers.
-    n_unique_kmers = hist.obs.sum()
+    n_unique_kmers = (hist['freq'] * hist['obs']).sum()
     logging.debug('n_unique_kmers %s', n_unique_kmers)
+    # case by / 0
```

Comment:
You could wrap in a `try/except/finally` or let the error propagate up and handle it by warning the user and setting the min kmer freq threshold to the default. It would be *really* weird if there are no unique kmers in a dataset, as in, you'd probably have a simulated dataset on your hands. Actual biology is very messy and error prone.